pkg/lib/queueinformer/queueinformer_operator.go

```diff
@@ -142,3 +147,34 @@ func (o *Operator) sync(loop *QueueInformer, key string) error {
 	}
 	return loop.syncHandler(obj)
 }
+
+func (o *Operator) handleMetrics(informer *QueueInformer) error {
+	switch informer.name {
+	case "csv":
```

Comment:
Instead of making the QueueInformer know about all of the informers we might construct it with, what if specialize `handleMetrics` per operator/control loop (e.g. CSV control loop triggers CSV metrics, InstallPlan control loop triggers InstallPlan metrics, etc) similar to how we have the operators specify a `syncHandler`.

Then when you make a new CSV QueueInformer (as a strawman) set the `metricHandler` to 

```go 
o.metricHandler := func() {
	cList, err := o.OpClient.ListCustomResource(v1alpha1.GroupName, v1alpha1.GroupVersion, "", v1alpha1.ClusterServiceVersionKind)
	if err != nil {
		return err
	}
	metrics.CSVCount.Set(float64(len(cList.Items)))
}
```

and `handleMetrics` just calls `o.metricHandler()`

(I think a more common go pattern would be to define a `MetricProvider` interface with a `MetricHandler()` method, but if you want to go that route you should update the syncHandler to follow the same pattern 😄)